### ISSUE TITLE
Misleading include subpaths

### ISSUE THREAD

**Issue by Anton-V-K** [ARMmbed/mbedtls#3778]
I tried to build part of the libary in Arduino IDE (for esp8266), and immediatey faced _strange_ include path in `mbedtls/md.h` like:
```
#include "mbedtls/config.h"
```
and then in standard `mbedtls/config.h`:
```
#include "mbedtls/check_config.h"
```

Fortunately the first case is easy to tune, so I have to include the libary this way:
```
#define MBEDTLS_CONFIG_FILE "config.h"
#include "mbedtls/md.h"
```
And I'll have to prepare custom `config.h` just to work around misleading subpath - `mbedtls/check_config.h`.

As far as I know the double quotes around an include path instructs the compiler to search from the current folder, which is meaningless in above mentioned cases, since there are no `mbedtls/mbedtls/config.h` or `mbedtls/mbedtls/check_config.h`.

Both these includes should be rewritten as 
```
#include "config.h"
```
and
```
#include "check_config.h"
```

Optionally (to maintain backward comparibitliy) the same trick may be employed as it is done for `config.h` in `md.h`:
```
#if !defined(MBEDTLS_CHECK_CONFIG_FILE)
#include "mbedtls/check_config.h"
#else
#include MBEDTLS_CHECK_CONFIG_FILE
#endif
```

**Comment by daverodgman** [comment]
It sounds like an include path problem - I think you need to add the directory named "include" to your include path. I'm not sure how this is done in the Arduino environment, though (modifying platform.txt seems to be one option). When this is done, includes like #include "mbedtls/config.h" should work properly.

**Comment by Anton-V-K** [comment]
Same or similar issue with include paths was already reported earlier as #2360 , and partially addressed in #2319. I wonder why the approach had been changed since then.

Anyway resolving this path problem for Arduino IDE is tricky, and I doubt the approach with hacking `platform.txt` is a way to go, since this file is user-specific (or machine-specific), while `mbedtls` is project-specific (may be cloned as a subproject).
